core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1384,14 +1383,61 @@ public boolean convertPartitionNativeToParquet(long partitionTimestamp) {
                     .$(", error=").$(e.getMessage()).I$();
 
             // rollback
-            if (!ff.rmdir(other.trimTo(newPartitionFolderLen).slash())) {
+            if (!ff.rmdir(other.trimTo(newPartitionDirLen).slash())) {
                 LOG.error().$("could not remove parquet file [path=").$(other).I$();
             }
             throw e;
         } finally {
             path.trimTo(pathSize);
             other.trimTo(pathSize);
         }
+
+        LOG.info().$("copying index files to parquet [path=").$substr(pathRootSize, path).I$();
+        try {
+            final int columnCount = metadata.getColumnCount();
+            for (int columnIndex = 0; columnIndex < columnCount; columnIndex++) {
```

Comment:
Can we merge the symbol column conversion and index building into the same loop iteration? Currently, we continue convert the partition and then have to roll it back later due to the index error.

Reply:
There is no common loop where we could do that. In the previous try block we mmap columns one by one and then encode the whole partition in a single `PartitionEncoder.encodeWithOptions()` call. If we move indexing into the upper loop, we'll simply change the encode-then-index order to index-then-encode, but these operations won't interleave. Does this make sense?